frontend/src/views/Join.tsx

```diff
@@ -49,15 +58,21 @@ function JoinGamePage() {
             }}
             onKeyPress={(event) => {
               if (event.key === 'Enter' && validNickname) {
-                connect(SOCKET_ENDPOINT, nickname);
-                setPageState(2);
+                connect(SOCKET_ENDPOINT, nickname, setUsers).then(() => {
+                  setPageState(2);
+                }).catch((response) => {
+                  setError(response.message);
```

Comment:
Building on my previous comment, if you decide to implement that change, then this might look something like `if isError(res) then setError((res as ErrorResponse).message) else setPageState(2)`